src/components/plots/barPlot.ts

```diff
@@ -142,6 +159,31 @@ export module Plots {
       }
     }
 
+    public addDataset(dataset: Dataset) {
+      if (this._isSetup) {
+        var labelArea = this._renderArea.append("g").classed(Bar._LABEL_AREA_CLASS, true);
```

Comment:
Well it would be more aligned to DOM nodes that are correlated to each `Dataset`